lib/src/booleans/boolean_contains.dart

```diff
@@ -0,0 +1,194 @@
+      return coords1 == coords2;
+    } else {
+      throw _exception();
+    }
+  } else if (geom1 is MultiPoint) {
+    if (geom2 is Point) {
+      return _isPointInMultiPoint(geom1, geom2);
+    } else if (geom2 is MultiPoint) {
+      return _isMultiPointInMultiPoint(geom1, geom2);
+    } else {
+      throw _exception();
+    }
+  } else if (geom1 is LineString) {
+    if (geom2 is Point) {
+      return booleanPointOnLine(geom2, geom1, ignoreEndVertices: true);
+    } else if (geom2 is LineString) {
+      return _isLineOnLine(geom1, geom2);
+    } else if (geom2 is MultiPoint) {
+      return _isMultiPointOnLine(geom1, geom2);
+    } else {
+      throw _exception();
+    }
+  } else if (geom1 is Polygon) {
+    if (geom2 is Point) {
+      return booleanPointInPolygon((geom2).coordinates, geom1,
+          ignoreBoundary: true);
+    } else if (geom2 is LineString) {
+      return _isLineInPoly(geom1, geom2);
+    } else if (geom2 is Polygon) {
+      return isPolyInPoly(geom1, geom2);
+    } else if (geom2 is MultiPoint) {
+      return _isMultiPointInPoly(geom1, geom2);
+    } else {
+      throw _exception();
+    }
+  } else {
+    throw _exception();
+  }
+}
+
+bool _isPointInMultiPoint(MultiPoint multiPoint, Point pt) {
+  for (int i = 0; i < multiPoint.coordinates.length; i++) {
+    if ((multiPoint.coordinates[i] == pt.coordinates)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+bool _isMultiPointInMultiPoint(MultiPoint multiPoint1, MultiPoint multiPoint2) {
+  for (Position coord2 in multiPoint2.coordinates) {
+    bool match = false;
+    for (Position coord1 in multiPoint1.coordinates) {
+      if (coord2 == coord1) {
+        match = true;
+      }
+    }
+    if (!match) return false;
+  }
+  return true;
+}
+
+bool _isMultiPointOnLine(LineString lineString, MultiPoint multiPoint) {
+  var haveFoundInteriorPoint = false;
+  for (var coord in multiPoint.coordinates) {
+    if (booleanPointOnLine(Point(coordinates: coord), lineString,
+        ignoreEndVertices: true)) {
+      haveFoundInteriorPoint = true;
+    }
+    if (!booleanPointOnLine(Point(coordinates: coord), lineString)) {
+      return false;
+    }
+  }
+  return haveFoundInteriorPoint;
+}
+
+bool _isMultiPointInPoly(Polygon polygon, MultiPoint multiPoint) {
+  for (var coord in multiPoint.coordinates) {
+    if (!booleanPointInPolygon(coord, polygon, ignoreBoundary: true)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+bool _isLineOnLine(LineString lineString1, LineString lineString2) {
+  var haveFoundInteriorPoint = false;
+  for (Position coord in lineString2.coordinates) {
+    if (booleanPointOnLine(
+      Point(coordinates: coord),
+      lineString1,
+      ignoreEndVertices: true,
+    )) {
+      haveFoundInteriorPoint = true;
+    }
+    if (!booleanPointOnLine(
+      Point(coordinates: coord),
+      lineString1,
+      ignoreEndVertices: false,
+    )) {
+      return false;
+    }
+  }
+  return haveFoundInteriorPoint;
```

Comment:
should `return true;` in the first `if` clause, and `return false;` at the end of the function. No need for the local `var haveFoundInteriorPoint`

Reply:
not applicable